### PR TITLE
feat(rust): support `file` option

### PR DESCRIPTION
**Pull request by hyf0**

### CODE REVIEW

**Comment by hyf0** [comment]
* **#2567** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/2567?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2565** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/2565?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#2564** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/2564?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2563** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/2563?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @hyf0 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: crates/rolldown/tests/rolldown/function/file/css/artifacts.snap [hunk 1]
```diff
@@ -0,0 +1,19 @@
+---
+source: crates/rolldown_testing/src/integration_test.rs
+snapshot_kind: text
+---
+# Assets
+
+## out.css
```
Comment (hyf0) [inline crates/rolldown/tests/rolldown/function/file/css/artifacts.snap:7]: This is not expected, we should have a error like this https://hyrious.me/esbuild-repl/?version=0.23.0&b=e%00entry.js%00import+%7B+parse+%7D+from+%27ext%27%0Aimport+%27.%2Ffoo%27%0Aimport+foo+from+%27.%2Ffile.txt%27%0Aconsole.log%28foo%29%0Aconsole.log%28parse%29&b=%00foo.css%00body+%7B%7D&b=%00file.txt%00console.log&b=%00file.js%00&o=--bundle+--tree-shaking%3Dfalse+--format%3Desm+--external%3Aext+--outfile%3Dout.css+--loader%3A.txt%3Dfile.

But, let's solve it in another PR.

**Comment by IWANABETHATGUY** [comment]
Since there is no context in your context, I assume the `file` corresponds to `outfile` in esbuild, any reason we don't align with esbuild ?

**Comment by IWANABETHATGUY** [comment]
> Since there is no context in your context, I assume the `file` corresponds to `outfile` in esbuild, any reason we don't align with esbuild ?

probably you mean `output.file` in  rollup, that make sense.

**Review by IWANABETHATGUY — APPROVED** [review]

**Comment by hyf0** [comment]
> Since there is no context in your context, I assume the `file` corresponds to `outfile` in esbuild, any reason we don't align with esbuild ?

Yeah. This is a tricky part. `outfile` is a more proper name. I was gonna use it in the first place. But using this name would cause the throwing error use the `outfile` too. It would be confusing, if we throw such error to node side.

I have some rough ideas on solving this problem, but it's too early to say.

**Comment by hyf0** [comment]
### Merge activity

* **Nov 3, 5:22 AM EST**: [Graphite](https://app.graphite.dev/github/pr/rolldown/rolldown/2565) rebased this pull request after merging its parent, because this pull request is set to merge when ready.

**Pull request merged**